scripts/globals/znm.lua

```diff
@@ -147,61 +251,326 @@ end
 xi.znm.ryo.onEventFinish = function(player, csid, option)
 end
 
+-----------------------------------
+-- Ryo General Helpers
+-----------------------------------
+xi.znm.ryo.tradedPlateValue = function(player)
+    return player:getLocalVar("[ZNM][Ryo]SoulPlateValue")
+end
+
+xi.znm.ryo.setTradedPlateValue = function(player, zeni)
+    player:setLocalVar("[ZNM][Ryo]SoulPlateValue", zeni)
+end
+
 -----------------------------------
 -- Sanraku
 -----------------------------------
 xi.znm.sanraku = xi.znm.sanraku or {}
 
-local platesTradedToday = function(player)
-    local currentDay = VanadielUniqueDay()
-    local storedDay = player:getCharVar("[ZNM][Sanraku]TradingDay")
+------------------------------------
+-- onTrade
+------------------------------------
+xi.znm.sanraku.onTrade = function(player, npc, trade)
+    if trade:getItemCount() == 1 then -- One soul plate or trophy at a time
+        local item = trade:getItem(0)
 
-    if currentDay ~= storedDay then
-        player:setCharVar("[ZNM][Sanraku]TradingDay", 0)
-        player:setCharVar("[ZNM][Sanraku]TradedPlates", 0)
-        return 0
+        if trade:getItemId() == xi.items.SOUL_PLATE then
+            xi.znm.sanraku.handleTradeWithPlate(player, npc, item)
+        else -- Check Trophy trading (for ZNM seals)
+            xi.znm.sanraku.handleTradeWithTrophy(player, npc, item)
+        end
     end
-
-    return player:getCharVar("[ZNM][Sanraku]TradedPlates")
 end
 
-xi.znm.sanraku.onTrade = function(player, npc, trade)
+------------------------------------
+-- onTrade Helpers
+------------------------------------
+xi.znm.sanraku.handleTradeWithPlate = function(player, npc, item)
     if not player:hasKeyItem(xi.ki.RHAPSODY_IN_AZURE) then
-        if platesTradedToday(player) >= 10 then
-            -- TODO: A message here?
+        local trade_limit = xi.znm.SOULPLATE_TRADE_LIMIT
+
+        if xi.znm.sanraku.platesTradedToday(player) >= trade_limit then
+            player:showText(npc, 13125, 1, xi.items.SOUL_PLATE,trade_limit)
             return
         end
     else -- If you have the KI, clear out the tracking vars!
-        player:setCharVar("[ZNM][Sanraku]TradingDay", 0)
-        player:setCharVar("[ZNM][Sanraku]TradedPlates", 0)
+        xi.znm.resetDailyTrackingVars(player)
     end
+    -- Cache the soulplate value on the player
+    local plateData = item:getSoulPlateData()
 
-    if npcUtil.tradeHasExactly(trade, xi.items.SOUL_PLATE) then
-        -- Cache the soulplate value on the player
-        local item = trade:getItem(0)
-        local plateData = item:getSoulPlateData()
-        player:setLocalVar("[ZNM][Sanraku]SoulPlateValue", plateData.zeni)
-        player:startEvent(910, plateData.zeni)
+    xi.znm.sanraku.setTradedPlateValue(player, plateData.zeni)
+    player:startEvent(910, plateData.zeni)
```

Comment:
I'll assume it was tested but one issue I ran into on wings was trying to align the 75 cap era zeni values per plate with the client side logic for the plate trade in CS.

In 75 cap era 100 points was exceptional.  However the client wings uses actually considers 100 points to be a bad example.  This is something that LSB likely does not have to be concerned with - but ASB will have to address

```
EventEntryPoint_16982568_910.L0B42:
.L0B42: MOV [0003], Param0
        Opcode20 01
        Opcode42
        LOOKAT [Player]
        WAIT16
        Opcode70
        MOV EventResult, #1
        UPDATEEVENT
        UPDATEEVENT WAIT
        NPCSAY #13135 // Oh my, is that a soul plate you have brought for me?\x07If I could be permitted to do the honors...<7F31>
        WAITFORINPUT
        NPCSAY #13136 // Aaah...hmmm...<7F31>
        WAITFORINPUT
        CMP [0003] >= #495, .L0B71
        NPCSAY #13139 // Hmmm... Surely you can do better than this, my friend. But I will accept it nevertheless.<7F31>
        WAITFORINPUT
        JMP .L0B84

.L0B71: CMP [0003] >= #990, .L0B80
        NPCSAY #13138 // A fine piece of photography!\x07You have my thanks!<7F31>
        WAITFORINPUT
        JMP .L0B84

.L0B80: NPCSAY #13137 // Magnificent! Truly a triumph in photography!\x07My faith in you is obviously well placed!<7F31>
        WAITFORINPUT
.L0B84: MOV Param1, [0003]
        PRINT #13087 // You earn {e:1} zeni.<7F31>
        WAITFORINPUT
        Opcode2E
        Opcode20 00
        FINISHEVENT
        END
```

Reply:
That makes sense.  When I was trying to wrap my head around zeni calculation factors - Nauls were the bane of my existence.  Once they were found by the player base no one cared about the fine details anymore.  It was Nauls or Nothing™ :joy:

Still, the client side handling of "how good" a picute is a pain and ASB should look into a client side modification.  On wings Sanraku is just perpetually unhappy.  I considered passing zeni*10 to the CS and displaying a custom string to players saying "you actually got 80 zeni, not 800" but in testing - people only read the number and didnt care if the old coot was happy or not.